code coverage is failed at some cases

``` console
KUBE_COVER="y" KUBE_COVERPROCS=8 ./hack/test-go.sh -- -p=2
--- FAIL: TestSnippetWriter (0.00s)
        snippet_writer_test.go:85: Expected "snippet_writer_test.go:78" but didn't find it in "template: /home/stack/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/libs/go2idl/generator  /snippet_writer_test.go:77:1: unclosed action"
FAIL
coverage: 10.2% of statements
FAIL    k8s.io/kubernetes/cmd/libs/go2idl/generator     0.053s
```

thanks @zhouhaibing089 !
Another issue is:
pkg/master/master_test.go
```
--- FAIL: TestThirdPartyDiscovery (1.01s)
        Error Trace:    master_test.go:982
                        master_test.go:955
        Error:          Not equal: unversioned.GroupVersionForDiscovery{GroupVersion:"company.com/v1", Version:"v1"} (expected)
                                != unversioned.GroupVersionForDiscovery{GroupVersion:"", Version:""} (actual)

                        Diff:
                        --- Expected
                        +++ Actual
                        @@ -1,4 +1,4 @@
                         (unversioned.GroupVersionForDiscovery) {
                        - GroupVersion: (string) (len=14) "company.com/v1",
                        - Version: (string) (len=2) "v1"
                        + GroupVersion: (string) "",
                        + Version: (string) ""
                         }

        Error Trace:    master_test.go:982
                        master_test.go:955
        Error:          Not equal: unversioned.GroupVersionForDiscovery{GroupVersion:"company.com/v3", Version:"v3"} (expected)
                                != unversioned.GroupVersionForDiscovery{GroupVersion:"", Version:""} (actual)

                        Diff:
                        --- Expected
                        +++ Actual
                        @@ -1,4 +1,4 @@
                         (unversioned.GroupVersionForDiscovery) {
                        - GroupVersion: (string) (len=14) "company.com/v3",
                        - Version: (string) (len=2) "v3"
                        + GroupVersion: (string) "",
                        + Version: (string) ""
                         }

I0429 15:59:47.079006   31130 genericapiserver.go:82] Adding storage destination for group
I0429 15:59:47.079106   31130 genericapiserver.go:82] Adding storage destination for group autoscaling
I0429 15:59:47.079148   31130 genericapiserver.go:82] Adding storage destination for group batch
I0429 15:59:47.079190   31130 genericapiserver.go:82] Adding storage destination for group extensions
I0429 15:59:48.036682   31130 tunneler.go:208] Private key /tmp/private-1461916787 exists, but public key does not
FAIL
coverage: 68.1% of statements
FAIL    k8s.io/kubernetes/pkg/master    15.152s
```

--- a/hack/test-go.sh
+++ b/hack/test-go.sh
@@ -255,7 +255,9 @@ runTests() {
   # must make sure the output from parallel runs is not mixed. To achieve this,
   # we spawn a subshell for each parallel process, redirecting the output to
   # separate files.
-  printf "%s\n" "${@}" | xargs -I{} -n1 -P${KUBE_COVERPROCS} \
+  # cmd/libs/go2idl/generator is fragile when run under coverage, so ignore it for now.
+  # see: https://github.com/kubernetes/kubernetes/issues/24967
+  printf "%s\n" "${@}" | grep -v "cmd/libs/go2idl/generator"| xargs -I{} -n1 -P${KUBE_COVERPROCS} \
     bash -c "set -o pipefail; _pkg=\"{}\"; _pkg_out=\${_pkg//\//_}; \
         go test ${goflags[@]:+${goflags[@]}} \
           ${KUBE_RACE} \

